File: np_platform_message_relay/android/src/main/kotlin/com/nkming/nc_photos/np_platform_message_relay/NpPlatformMessageRelayPlugin.kt
package com.nkming.nc_photos.np_platform_message_relay

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodChannel

class NpPlatformMessageRelayPlugin : FlutterPlugin {
	override fun onAttachedToEngine(
		@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding
	) {
		val handler = MessageRelayChannelHandler()
		eventChannel = EventChannel(
			flutterPluginBinding.binaryMessenger,
			MessageRelayChannelHandler.EVENT_CHANNEL
		)
		eventChannel.setStreamHandler(handler)
		methodChannel = MethodChannel(
			flutterPluginBinding.binaryMessenger,
			MessageRelayChannelHandler.METHOD_CHANNEL
		)
		methodChannel.setMethodCallHandler(handler)
	}

	override fun onDetachedFromEngine(
		@NonNull binding: FlutterPlugin.FlutterPluginBinding
	) {
		eventChannel.setStreamHandler(null)
		methodChannel.setMethodCallHandler(null)
	}

	private lateinit var eventChannel: EventChannel
	private lateinit var methodChannel: MethodChannel
}

File: np_platform_message_relay/ios/Classes/NpPlatformMessageRelayPlugin.swift
import Flutter
import UIKit

public class NpPlatformMessageRelayPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let handler = MessageRelayChannelHandler()
    let eventChannel = FlutterEventChannel(
      name: MessageRelayChannelHandler.eventChannel,
      binaryMessenger: registrar.messenger()
    )
    eventChannel.setStreamHandler(handler)
    let methodChannel = FlutterMethodChannel(
      name: MessageRelayChannelHandler.methodChannel,
      binaryMessenger: registrar.messenger()
    )
    methodChannel.setMethodCallHandler(handler.onMethodCall)
  }
}
